src/common/snippets/src/lowered/pass/mark_invariant_shape_path.cpp

```diff
@@ -0,0 +1,130 @@
+// Copyright (C) 2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+
+#include "snippets/lowered/pass/mark_invariant_shape_path.hpp"
+
+#include "snippets/lowered/expressions/buffer_expression.hpp"
+#include "snippets/op/memory_access.hpp"
+#include "snippets/snippets_isa.hpp"
+#include "snippets/utils/utils.hpp"
+#include "snippets/itt.hpp"
+
+namespace ov {
+namespace snippets {
+namespace lowered {
+namespace pass {
+
+namespace {
+
+// Specific value to mark ports which does't affect output shape of broadcastable ops.
+// For example, ops with output scalar shape or Horizon ops.
+static const size_t NOT_AFFECTING_PATH = SIZE_MAX;
+
+static bool is_shape_broadcastable_op(const ExpressionPtr& expr) {
+    const auto& node = expr->get_node();
+    return ov::op::util::is_binary_elementwise_arithmetic(node) ||
+           ov::op::util::is_binary_elementwise_comparison(node) ||
+           ov::op::util::is_binary_elementwise_logical(node) ||
+           ov::is_type<ov::op::v1::Select>(node);
+}
```

Comment:
BTW `ov::Node` has virtual `get_autob()` method: it returns `NONE` by default, but is overridden in all the ops which mentioned here: binary eltwises and select. Maybe we can call this method and check the result? Such general check will allow to avoid explicit op types enumeration.
I don't think that this is really necessary though, so it's up to you